dts/bindings/base/base.yaml

```diff
@@ -38,3 +38,20 @@ properties:
         type: array
         category: optional
         description: Clock gate information
+
+    # pinctrl-0 and pinctrl-names appear in .dts files but are unused as of
+    # writing. Declare them here just so that we can require that all device
+    # node properties are declared in bindings.
+    #
+    # See commit b023fbf938 ("dts/bindings: Remove pinctrl from bindings") as
+    # well.
+
+    pinctrl-0:
```

Comment:
Used to be "pinctrl-/d+:" for a regular expression. Maybe better to remove the unused pinctrl definitions in the DTS than to introduce a only partly useful binding definition.

Reply:
Looks like there's a lot of them, so was afraid I'd remove useful information.

Fine with me to remove them though, but not up to me.

Reply:
Pinctrl info is for shure valuable. In this case the binding should account for the generic case "pinctrl-/d+:". This requires changes in edtslib.py.

Reply:
How complete is the info at the moment?

OTOH, if it's not used, and it would be easy to add back, then it would be better to remove it from the DTS files. No way to know that it's unused just from looking at the DTS files now.

Reply:
I´m using some of the pinctrl info in my project. But im atypical as I´m using code generation and can easily add the used pinctrl-x info to the project DTS. I already patched extract_dts_includes to not bail out on pinctrl-x and will do the same for edtslib. So from my side you may well remove it from DTS.

I think @galak knows most about that and should decide on the proper solution.